renku/core/metadata/repository.py

```diff
@@ -423,6 +425,65 @@ def get_content_from_submodules():
 
         raise errors.ExportError(f"File not found in the repository: '{revision}/{checksum}:{path}'")
 
+    def get_object_hashes(self, paths: List[Union[Path, str]], revision: str = None) -> Dict[str, str]:
+        """Return git hash of an object in a Repo or its submodule.
+
+        NOTE: path must be relative to the repo's root regardless if this function is called from a subdirectory or not.
+        """
+        absolute_paths = set(get_absolute_path(path, self.path) for path in paths)
+
+        hashes = {}
+        # NOTE: If revision is not specified, we use hash-object to hash the (possibly) modified object
+        if not revision:
+            staged_files = [d.a_path for d in self.staged_changes] if self.head.is_valid() else []
+            modified_files = [item.b_path for item in self.unstaged_changes if item.b_path]
+            dirty_files = {os.path.join(self.path, p) for p in self.untracked_files + modified_files + staged_files}
+            dirty_files = {p for p in dirty_files if p in absolute_paths and not os.path.isdir(p)}
+            clean_files = {p for p in absolute_paths if p not in dirty_files}
+            dirty_files = list(dirty_files)
+
+            dirty_files_hashes = Repository.hash_objects(dirty_files)
+            hashes.update(zip(dirty_files, dirty_files_hashes))
+
+            if not clean_files:
+                return hashes
+            revision = "HEAD"
+            absolute_paths = clean_files
+
+        submodule_paths = defaultdict(list)
+        main_repo_paths = []
+
+        if self.submodules:
+            for absolute_path in absolute_paths:
+                for submodule in self.submodules:
+                    try:
+                        Path(absolute_path).relative_to(submodule.path)
+                        submodule_paths[submodule].append(absolute_path)
+                    except ValueError:
+                        main_repo_paths.append(os.path.relpath(absolute_path, start=self.path))
+        else:
+            main_repo_paths = map(lambda p: os.path.relpath(p, start=self.path), absolute_paths)
+
+        if main_repo_paths:
+            main_repo_paths = Repository.list_paths_in_revision(
+                main_repo_paths, revision=revision, working_dir=self.path
+            )
```

Comment:
The reason to call `list_paths_in_revision` is to make sure that paths exists in the repo (to make rev-parse not failing later), right? If so, can we. use a better name for it to make this more. clear?